src/sentry/models/integrations/organization_integration.py

```diff
@@ -56,3 +56,59 @@ def delete(self, *args, **kwds):
             for outbox in self.outboxes_for_update():
                 outbox.save()
             super().delete(*args, **kwds)
+
+    @staticmethod
+    def services_in(config: Mapping[str, Any]) -> List[PagerDutyServiceDict]:
+        return config.get("pagerduty_services", [])
+
+    def set_services(self, services: List[PagerDutyServiceDict]) -> None:
+        self.config["pagerduty_services"] = services
+
+    @staticmethod
+    def find_service(config: Mapping[str, Any], id: int | str) -> PagerDutyServiceDict | None:
+        try:
+            return next(
+                pds
+                for pds in OrganizationIntegration.services_in(config)
+                if str(pds["id"]) == str(id)
+            )
+        except StopIteration:
+            return None
+
+    def add_pagerduty_service(
+        self, integration_key: str, service_name: str
+    ) -> PagerDutyServiceDict:
+        from sentry.models import OrganizationIntegration
+
+        with transaction.atomic(router.db_for_write(OrganizationIntegration)):
+            OrganizationIntegration.objects.filter(id=self.id).select_for_update()
+
+            with transaction.get_connection(
+                router.db_for_write(OrganizationIntegration)
+            ).cursor() as cursor:
+                cursor.execute(
+                    "SELECT nextval(%s)", [f"{OrganizationIntegration._meta.db_table}_id_seq"]
+                )
+                next_id: int = cursor.fetchone()[0]
+
+            service: PagerDutyServiceDict = {
+                "id": next_id,
+                "integration_key": integration_key,
+                "service_name": service_name,
+                "integration_id": self.integration_id,
+            }
+
+            existing: list[PagerDutyServiceDict] = OrganizationIntegration.services_in(self.config)
```

Comment:
Do you need this annotation if the return of `services_in` is also typed?

Reply:
Not strictly, but it was helpful reading to contextualize esp because it previously was an ORM but now it is a dict.  I simply considered it a "helpful" annotation that helped me understand what the code is doing and what the types are at a glance.

To be clear, is there any official policy on code style wrt to explicit annotations?  I made this change with good intention for readibility, and I'm unaware if we have (or should have) any linters or official policy.